src/external/token/ERC20Issuance_Blacklist_v1.sol

```diff
@@ -0,0 +1,214 @@
+ *
+ * @notice  An ERC20 token implementation that extends ERC20Issuance_v1 with
+ *          blacklisting capabilities. This allows accounts with the blacklist manager role to restrict specific
+ *          addresses from participating in token operations.
+ *
+ * @dev     This contract inherits from:
+ *              - IERC20Issuance_Blacklist_v1
+ *              - ERC20Issuance_v1
+ *          Key features:
+ *              - Individual address blacklisting.
+ *              - Batch blacklisting operations.
+ *              - Owner-controlled manager role assignment.
+ *              - Blacklist manager controlled blacklist management.
+ *              Blacklist operations are performed by accounts with the blacklist manager role,
+ *              while the contract owner controls who can be a blacklist manager.
+ *          All blacklist operations can only be performed by accounts with the blacklist manager role.
+ *
+ * @custom:security-contact security@inverter.network
+ *                          In case of any concerns or findings, please refer to
+ *                          our Security Policy at security.inverter.network or
+ *                          email us directly!
+ *
+ * @custom:version  1.0.0
+ *
+ * @custom:standard-version 1.0.0
+ *
+ * @author  Zealynx Security
+ */
+contract ERC20Issuance_Blacklist_v1 is
+    IERC20Issuance_Blacklist_v1,
+    ERC20Issuance_v1
+{
+    // --------------------------------------------------------------------------
+    // Constants
+
+    /// @notice	Maximum number of addresses that can be blacklisted in a batch.
+    uint public constant BATCH_LIMIT = 200;
+
+    // --------------------------------------------------------------------------
+    // Storage
+
+    /// @notice	Mapping of blacklisted addresses.
+    mapping(address account => bool isBlacklisted) private _blacklist;
+
+    /// @notice	Mapping of blacklist manager addresses.
+    mapping(address account => bool isManager) private _isBlacklistManager;
+
+    // --------------------------------------------------------------------------
+    // Modifiers
+
+    /// @notice Modifier to check if the caller is a blacklist manager.
+    modifier onlyBlacklistManager() {
+        if (!_isBlacklistManager[_msgSender()]) {
+            revert ERC20Issuance_Blacklist_NotBlacklistManager();
+        }
+        _;
+    }
+
+    // --------------------------------------------------------------------------
+    // Constructor
+
+    /// @notice	Constructor for ERC20Issuance_Blacklist_v1.
+    /// @param	name_ Token name.
+    /// @param	symbol_ Token symbol.
+    /// @param	decimals_ Token decimals.
+    /// @param	initialSupply_ Initial token supply.
+    /// @param	initialAdmin_ Initial admin address.
+    /// @param	initialBlacklistManager_ Initial blacklist manager (typically an EOA).
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        uint8 decimals_,
+        uint initialSupply_,
```

Comment:
```suggestion
        uint maxSupply_,
```

Reply:
done

Reply:
oh shoot, I pushed it to our test branch, I'll push it to this branch right away

Reply:
done